rubble-nrf5x/src/radio.rs

```diff
@@ -83,14 +86,61 @@ pub struct BleRadio {
 
 impl BleRadio {
     /// Initializes the radio in BLE mode and takes ownership of the RX and TX buffers.
-    // TODO: Use type-safe clock configuration to ensure that chip uses ext. crystal
+    #[cfg(feature = "51")]
     pub fn new(
         radio: RADIO,
+        ficr: pac::FICR,
```

Comment:
Since FICR is read-only, it should suffice to pass a reference to it instead of taking ownership (which would make it inaccessible to other code).

It might even be a good idea to take that reference even for the nRF52 chips even though it isn't needed, in order to make the API more uniform (and passing an additional reference isn't really much of a burden to the user). It would allow getting rid of the `#[cfg]`s.